waspc/examples/todoApp/ext/Todo.js

```diff
@@ -98,12 +99,26 @@ const Tasks = (props) => {
 }
 
 const Task = (props) => {
+  const action = useAction(updateTaskIsDone, {
+    optimisticUpdates: [{
+      getQuery: () => [getTasks],
+      // TODO: decide what we do when the cache
+      updateQuery: (updatedTask, oldTasks) => {
```

Comment:
How should we behave when updating a query that does not yet exist in the cache?

Such a scenario can happen in the rare scenario when the users/developers wish to optimistically update something that is not in the current view. The `oldData` will in this case be `undefined`.

We can just document that the `updateQuery` should be ready to handle the case when `oldTasks` is `undefined`. This approach does seem a bit janky, though.

Another option is ignoring such queries (i.e., not performing optimistic updates on queries not yet in the cache). This strategy would probably be OK in terms of UX (the query will get updated once the user lands on a view that features it, but it's less than ideal from the developer's perspective (a dev thinks they're updating something, and we're silently dropping it). We could mitigate the confusion by emphasizing that the `updateFn` _will not_ get called unless the provided query is already in the cache and optionally make this even more explicit by changing the function's name to something like `updateQueryIfCached`. That might be a bit extreme, though.

Reply:
Good idea. I'll go with it, seems like to most reasonable approach.